packages/collabswarm/src/keychain.ts

```diff
@@ -0,0 +1,36 @@
+/**
+ * A keychain contains a CollabswarmDocument's encryption keys.
+ *
+ * @tparam KeychainChange Type of a change record describing edits made to the document keychain.
+ * @tparam DocumentKey Type of a document encryption key.
+ */
+export interface Keychain<KeychainChange, DocumentKey> {
+  /**
+   * Add a document encryption key to the keychain.
+   *
+   * @param key Document encryption key to add.
+   * @return A change record describing the keychain addition.
+   */
+  add(key: DocumentKey): Promise<KeychainChange>;
+
+  /**
+   * Gets a change record describing the whole state of the keychain.
```

Comment:
I’m expecting this to give a list of changes, how can one record be the entire history? So again maybe I’m missing the definition, but above it implies ‘edits’ in one change…actually now I’m not sure how it would be plural, sure for ACL adding 2 users, but a document key changes at most once at each change

Reply:
Updated the name/comment to be more descriptive